test/Microsoft.Diagnostics.EventFlow.Outputs.Tests/ElasticSearchOutputTests.cs

```diff
@@ -86,9 +88,38 @@ public void VerifyAllConfigOptionsMappedToConfigObject()
                 Assert.Equal("verysecret", esOutputConfiguration.Proxy.UserPassword);
 
                 Assert.Equal("date_nanos", esOutputConfiguration.Mappings.Properties["timestamp"].Type);
+
+                var eso = new ElasticSearchOutput(esOutputConfiguration, new CustomHealthReporter(configuration));
+                Assert.Equal($"myprefix-{DateTimeOffset.UtcNow:yyyy.MM}", eso.GetIndexName(eso.connectionData));
             }
         }
 
+        [Theory]
+        [InlineData(2023, 1, 1, "yyyy.q", "2023.q1")]
+        [InlineData(2023, 3, 1, "yyyy.q", "2023.q1")]
+        [InlineData(2023, 4, 1, "yyyy.q", "2023.q2")]
+        [InlineData(2023, 1, 1, "yyyy.MM.w", "2023.01.w1")]
+        [InlineData(2023, 2, 1, "yyyy.MM.w", "2023.02.w1")]
```

Comment:
Please add a test for week-of-month that is not the first week ☺️

Reply:
Yah, that is what let the previous bug get by. :-D